query-engine/prisma/src/tests/type_mappings/test_api.rs

```diff
@@ -0,0 +1,211 @@
+use super::super::test_api::QueryEngine;
+use crate::context::PrismaContext;
+use quaint::prelude::Queryable;
+
+pub type TestResult = anyhow::Result<()>;
+
+pub struct TestApi {
+    provider: &'static str,
+    database_string: String,
+    is_pgbouncer: bool,
+}
+
+impl TestApi {
+    fn datasource(&self) -> String {
+        format!(
+            r#"
+                datasource my_db {{
+                    provider = "{provider}"
+                    url = "{url}"
+                }}
+            "#,
+            provider = self.provider,
+            url = self.database_string,
+        )
+    }
+
+    pub async fn execute(&self, sql: &str) -> anyhow::Result<()> {
+        let conn = quaint::single::Quaint::new(&self.database_string).await?;
```

Comment:
If we already have `&self` here why not just opening one connection to the instance to speed up the tests? Can also be lazy...

Reply:
yes I went with the easy solution, I can change that

Reply:
hmm we only ever call `execute` once, so I don't know if it's worth optimizing. We'd need to change the test macro or use a mutex/oncecell too, since we can't take `TestApi` by `&mut`.

Reply:
Once cell is complicated because `Quaint::new` is async.